File: internal/indexer/testdata/ts/sample.ts
import { EventEmitter } from "events";

const MAX_RETRIES = 3;
let defaultName = "world";

interface Greeter {
  greet(name: string): string;
}

type PersonOptions = {
  name: string;
  age: number;
};

enum Status {
  Active = "active",
  Inactive = "inactive",
}

export class Person implements Greeter {
  name: string;
  age: number;

  constructor(options: PersonOptions) {
    this.name = options.name;
    this.age = options.age;
  }

  greet(name: string): string {
    return `Hello, ${name}! I'm ${this.name}.`;
  }

  get displayName(): string {
    return this.name.toUpperCase();
  }
}

export function sayHello(): void {
  const p = new Person({ name: defaultName, age: 30 });
  const msg = p.greet("friend");
  console.log(msg);
  for (let i = 0; i < MAX_RETRIES; i++) {
    console.log(p.greet(defaultName));
  }
  console.log(p.displayName);
}

export const formatName = (name: string): string => {
  return name.trim();
};

namespace Utils {
  export function helper(): void {}
}

File: internal/indexer/testdata/rust/sample.rs
use std::fmt;

const MAX_RETRIES: u32 = 3;

pub struct Person {
    pub name: String,
    pub age: u32,
}

pub enum Status {
    Active,
    Inactive,
}

pub trait Greeter {
    fn greet(&self, name: &str) -> String;
}

impl Greeter for Person {
    fn greet(&self, name: &str) -> String {
        format!("Hello, {}! I'm {}.", name, self.name)
    }
}

impl Person {
    pub fn new(name: String, age: u32) -> Self {
        Person { name, age }
    }

    pub fn say_hello(&self) {
        let msg = self.greet("friend");
        println!("{}", msg);
        for _ in 0..MAX_RETRIES {
            println!("{}", self.greet(&self.name));
        }
    }
}

mod helpers {
    pub fn format_name(name: &str) -> String {
        name.to_uppercase()
    }
}

type PersonAlias = Person;

fn main() {
    let p = Person::new("Alice".to_string(), 30);
    p.say_hello();
    let formatted = helpers::format_name(&p.name);
    println!("{}", formatted);
}
